### DISCUSSION TITLE
p-fileUpload Validation

### DISCUSSION THREAD

**Discussion by lm-swaroop** [primefaces/community#1018] [PrimeNG] [answered]
Hi,
I'm using ```p-fileUpload``` in multiple mode and I want to validate files, if file fails validation check it shouldn't be added to the list of files to be uploaded. I could not find any way to achieve this in the p-upload documentation. Like the ```onBeforeUpload``` event I need ```onBeforeSelect``` how could I achieve this?

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

This feature is not supported yet. However, a workaround would be possible by utilizing onSelect emitter. You can validate the selection when it is selected and throw an error depending on it.
